src/material/core/option/option.ts

```diff
@@ -107,10 +108,10 @@ export class MatOption<T = any> implements FocusableOption, AfterViewChecked, On
   /** Whether the option is disabled. */
   @Input({transform: booleanAttribute})
   get disabled(): boolean {
-    return (this.group && this.group.disabled) || this._disabled;
+    return (this.group && this.group.disabled) || this._disabled();
```

Comment:
Should we migrate the group as well and turn this into a `computed`?

Reply:
Hmmm, that's a good thought. It would either require a "breaking" change to the public API of the group or to add a getter/setter pair there so I can set an internal signal.